### PR TITLE
feat: update sql builder to use flattened  tables and MVs

### PR DESCRIPTION
**Pull request by llmin**
----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

## Summary

update sql builder to use flattened  tables and MVs.

## Implementation highlights

(describe how the merge request does for feature changes, share the RFC link if it has)

## Test checklist

- [x] add new test cases
- [x] all code changes are covered by unit tests
- [ ] end-to-end tests
  - [ ] deploy web console with CloudFront + S3 + API gateway
  - [ ] deploy web console within VPC
  - [ ] deploy ingestion server
    - [ ] with MSK sink
    - [ ] with KDS sink
    - [ ] with S3 sink
  - [ ] deploy data processing
  - [ ] deploy data modeling
    - [ ] new Redshift Serverless
    - [ ] provisioned Redshift
    - [ ] Athena
  - [ ] deploy with reporting

## Is it a breaking change

- [ ] add parameters without default value in stack
- [ ] introduce new service permission in stack
- [ ] introduce new top level stack module

## Miscellaneous

- [ ] introduce new symbol link source file(s) to be shared among infra code, web console frontend, and web console backend

### CODE REVIEW

File: src/control-plane/backend/lambda/api/service/quicksight/sql-builder.ts [hunk 1]
```diff
@@ -163,207 +162,343 @@ const baseColumns = `
...
+  const eventNameClause = eventNames.length > 0 ? eventNameInClause : '';
+
+  if (isNodePathSQL) {
+    return `
+    and event.event_name = '${ (sqlParameters.pathAnalysis?.platform === MetadataPlatform.ANDROID || sqlParameters.pathAnalysis?.platform === MetadataPlatform.IOS) ? '_screen_view' : '_page_view' }'
+    ${sqlParameters.pathAnalysis!.platform ? 'and platform = \'' + sqlParameters.pathAnalysis!.platform + '\'' : '' }
+    `;
+  } else if (isEventPathSQL) {
+    return 'and event.event_name not in (\'' + builtInEvents.filter(event => !eventNames.includes(event)).join('\',\'') + '\')';
+  } else {
```
Comment (zxkane) [inline src/control-plane/backend/lambda/api/service/quicksight/sql-builder.ts:280]: remove redundant else

File: src/control-plane/backend/lambda/api/service/quicksight/sql-builder.ts [hunk 2]
```diff
@@ -163,207 +162,343 @@ const baseColumns = `
...
     }
 
-    conditionSql = conditionSql.concat(`
-      ${index === 0? '' : 'or' } ( event_name = '${eventNameAndSQLCondition.eventName}' ${normalConditionSql} ${nestSqlPair0} )
-    `);
+    const userCondition = _hasUserCondition(sqlParameters);
+    let userJoinTable = '';
+    let userColList: string[] = [];
+    if (userCondition.hasUserAttribute) {
+      userJoinSql = `join ${sqlParameters.schemaName}.clickstream_user_attr_view as user_param
```
Comment (zxkane) [inline src/control-plane/backend/lambda/api/service/quicksight/sql-builder.ts:322]: We have `user-m-view` built by new transformer. Should you use the new view or update the user_attr view to use it?

**Review by zxkane — CHANGES_REQUESTED** [review]

File: src/control-plane/backend/lambda/api/service/quicksight/sql-builder.ts (hunk 2 above)
Comment (llmin) [reply src/control-plane/backend/lambda/api/service/quicksight/sql-builder.ts:322]: all sql is build on origin table and MVs in this PR.

File: src/control-plane/backend/lambda/api/service/quicksight/sql-builder.ts [hunk 3]
```diff
@@ -163,207 +162,343 @@ const baseColumns = `
...
+    and event.event_name not in ('${builtInEvents.filter(event => !eventNames.includes(event)).join('\',\'')}')
+    and event_param.event_parameter_key = '${nodeType}'
+    group by 1,2
+  `;
+}
+
+function _buildCommonColumnsSql(columns: string[], key: string, value: string) {
+  let columnsSql = '';
+  const columnList: string[] = [];
+  for ( const [_index, col] of columns.entries()) {
```
[the remark this answers is not in the document]
Comment (llmin) [reply src/control-plane/backend/lambda/api/service/quicksight/sql-builder.ts:221]: fixed

File: src/control-plane/backend/lambda/api/service/quicksight/sql-builder.ts (hunk 1 above)
Comment (llmin) [reply src/control-plane/backend/lambda/api/service/quicksight/sql-builder.ts:280]: fixed

File: src/control-plane/backend/lambda/api/service/quicksight/sql-builder.ts (hunk 2 above)
Comment (llmin) [reply src/control-plane/backend/lambda/api/service/quicksight/sql-builder.ts:322]: hold on this PR， will do more test on new tables.

**Pull request closed without merging**